tests/net/socket/getaddrinfo/src/main.c

```diff
@@ -47,6 +250,7 @@ void test_main(void)
 	k_thread_system_pool_assign(k_current_get());
 
 	ztest_test_suite(socket_getaddrinfo,
+			 ztest_user_unit_test(test_getaddrinfo_setup),
```

Comment:
if this is the only scenario that doesn't work in user mode, instantiate it with "ztest_unit_test" instead..

Reply:
Ah, I did not know that. Thanks for the info, will try that then.

Reply:
@jukkar it seems you have done this for other tests, disabling userspace completely for tests/net/getnameinfo even though it works in user mode.

Please do not do this again without consulting me, disabling user mode tests for APIs which are supposed to be tested under user mode will just lead to bit-rot and unwelcome surprises later.

Reply:
Also, I'm seeing there are new socket APIs without system calls. I'm fixing what I can, but please reject in code review any new BSD Socket API implementations that don't work in user mode.

Reply:
Yep, good point, will do.